lib/clients/platform-api/client.js

```diff
@@ -0,0 +1,33 @@
+const NYPLDataApiClient = require('@nypl/nypl-data-api-client')
+
+const kms = require('../kms.js')
+
+let clientPromise = null
+
+/**
+ * Initialize a Platform API client from encrypted creds found in process.env
+ *
+ */
+const init = async () => {
+  // Create a Promise to decrypt creds and resolve a client:
+  if (!clientPromise) {
+    const [key, secret] = await Promise.all([
+      kms.decrypt(process.env.NYPL_OAUTH_KEY),
+      kms.decrypt(process.env.NYPL_OAUTH_SECRET)
+    ])
+
+    clientPromise = new NYPLDataApiClient({
+      base_url: process.env.NYPL_API_BASE_URL,
+      oauth_key: key,
+      oauth_secret: secret,
+      oauth_url: process.env.NYPL_OAUTH_URL,
+      log_level: 'error'
+    })
+  }
+
+  return clientPromise
+}
+
+module.exports = {
+  init
+}
```

Comment:
I'm not sure about the division between "clients" and "things that aren't clients". If I'm looking for how the app integrates KMS, I'd think twice about whether that's a "client" or not, so I'm not sure placing modules that have client-like responsibilities (i.e. interface external apis) into a `clients` folder helps. It also creates a problem in that it's unclear what to do with functions that leverage client functions, but aren't strictly part of the client (e.g. all of the things in `get-fors.js`). I think subfoldering the `lib` folder is fine, but maybe just `lib/platform-api` (containing `client.js` and `get-fors.js`) for example. (Separately, I like placing convenience functions in `get-fors` but I'm not sure that's the best name. Maybe just `utils` (i.e. `lib/platform-api/utils.js`)?